server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -420,14 +420,16 @@ protected ValuesSourceAggregatorFactory innerBuild(
         AggregatorFactory parent,
         AggregatorFactories.Builder subFactoriesBuilder
     ) throws IOException {
-        if (IndexMode.TIME_SERIES.getName().equals(context.getIndexSettings().getSettings().get(IndexSettings.MODE.getKey()))
-            && DateIntervalWrapper.IntervalTypeEnum.CALENDAR.equals(dateHistogramInterval.getIntervalType())) {
+        final String indexMode = context.getIndexSettings().getSettings().get(IndexSettings.MODE.getKey());
+        final DateIntervalWrapper.IntervalTypeEnum dateHistogramIntervalType = dateHistogramInterval.getIntervalType();
+        if (IndexMode.TIME_SERIES.getName().equals(indexMode)
+            && DateIntervalWrapper.IntervalTypeEnum.CALENDAR.equals(dateHistogramIntervalType)) {
             throw new UnsupportedAggregationOnDownsampledField(
```

Comment:
Maybe we would like to use another exception here which describes better the issue. I reused the existing that I used for the other error (fail on unsupported aggregations). Something like `UnsupportedDateHistogramIntervalOnTimeSeriesIndex`?